### PR TITLE
Towards Version 0.2.0

### PR DESCRIPTION
**Pull request by c-bebop**
This PR shall help identify issues and small further improvements to the code working towards the release of version 0.2.0.

### CODE REVIEW

File: include/gdsb/graph_io.h [hunk 1]
```diff
@@ -68,13 +77,35 @@ constexpr bool full_graph = false;
 //! @param  subgraph        A subgraph to extract from the file. Use default or
 //!                         any other Subgraph object if not specified by
 //!                         ExtractSubgraph.
-template <typename Vertex, typename EmplaceF, bool IsDirected, bool IsWeighted, bool IsDynamic = false, typename Timestamp = uint64_t, bool ExtractSubgraph = false>
```
Comment (fidus58) [inline include/gdsb/graph_io.h]: a type safe graph description could go along

```
#include <iostream>

template <bool b> class Property{
public: static constexpr bool is(){return b;}
};

class Directed: private Property<true>{
public: static constexpr bool isDirected(){return Property::is();}  
};
class Undirected: private Property<false>{
public: static constexpr bool isDirected(){return Property::is();}  
};
class Weighted: private Property<true>{
public: static constexpr bool isWeighted(){return Property::is();}  
};
class Unweighted: private Property<false>{
public: static constexpr bool isWeighted(){return Property::is();}  
};
class Dynamic: private Property<true>{
public: static constexpr bool isDynamic(){return Property::is();}  
};
class Static: private Property<false>{
public: static constexpr bool isDynamic(){return Property::is();}  
};
template <typename Directed, 
          typename Weighted, 
          typename Dynamic> class InputGraph {
        public:
        static constexpr bool isDirected() {return Directed::isDirected();}
        static constexpr bool isWeighted() {return Weighted::isWeighted();}
        static constexpr bool isDynamic()  {return Dynamic::isDynamic();}
};

int main() {

    std::cout<< InputGraph<Directed, Weighted, Static>::isDirected()<<"\n";
    //std::cout<< InputGraph<Weighted, Directed, Static>::isDirected()<<"\n";   
}

```
or a condensed version of it, i know 'much ado about little' but not heavier in the end

**Review by fidus58 — COMMENTED** [review]
not more so far

**Review by fidus58 — APPROVED** [review]
fine now.
only l. 78 reads a bit weird: 
`...typename Directed = Undirected, typename Weighted = Unweighted, typename Dynamic = Static ...`
but with the using declarations for the client ok

**Comment by c-bebop** [comment]
> fine now. only l. 78 reads a bit weird: `...typename Directed = Undirected, typename Weighted = Unweighted, typename Dynamic = Static ...` but with the using declarations for the client ok

Ah yes, maybe I'll change that at some point but I do want to setup the default parameters as such. Maybe renaming the template parameters could help.

**Comment by c-bebop** [comment]
Thanks @fidus58 for the feedback and help!

**Pull request merged**